integration/spark-interfaces-scala/src/main/scala/io/openlineage/spark/builtin/scala/v1/planNodes.scala

```diff
@@ -0,0 +1,112 @@
+/** Copyright 2018-2024 contributors to the OpenLineage project
+  * SPDX-License-Identifier: Apache-2.0
+  */
+package io.openlineage.spark.builtin.scala.v1
+
+import io.openlineage.client.OpenLineage.{
+  DatasetFacetsBuilder,
+  InputDatasetInputFacetsBuilder,
+  OutputDatasetOutputFacetsBuilder
+}
+import io.openlineage.client.utils.DatasetIdentifier;
+
+/** Trait to be implemented for LogicalPlan nodes to extract lineage information
+  * about input datasets.
+  */
+trait InputLineageNode {
+
+  /** Gets input dataset read by this LogicalPlans node
```

Comment:
Please describe those more from perspective of a `LineagePlan` developer - when to implement those, what assumptions do you have like in which Spark lifecycle part are you going to call those - or point to docs where this is explained. 

The comments in the current form make sense for people who would call those methods, not for those who would have to implement those.